typed-racket-lib/typed-racket/private/type-contract.rkt

```diff
@@ -526,6 +526,11 @@
        [(? Fun? t) (t->sc/fun t)]
        [(? DepFun? t) (t->sc/fun t)]
        [(Set: t) (set/sc (t->sc t))]
+       [(Sequence: (list t))
+        #:when (subtype t:-Nat t)
+        ;; sequence/c is always a wrapper, so avoid it when we just have a number
+        (or/sc (flat/sc #'exact-nonnegative-integer?)
```

Comment:
We should merge #631 first, because with this change and without that PR, every `(Sequence T)` type in a "trusted positive position" is going to optimize to an `any/c` contract.

```
(or/sc (flat/sc ....) (sequence/sc ....))
=>
(or/sc any/c (sequence/sc ....))
=>
any/c
```

Reply:
(Even though sequences are read-only, this is a problem if `T` is something like `(Boxof Int)`)